src/libcaf.h

```diff
@@ -89,7 +91,6 @@ typedef enum caf_deregister_t {
 caf_deregister_t;
 
 typedef void* caf_token_t;
-#ifdef GCC_GE_7
```

Comment:
Whats the danger of exposing them? Some things change interfaces and semantics, which we need to be careful of, but if the library is providing a new symbol or function that isn't used by the older compiler is there any danger?